### PR TITLE
Add informal definition of "label" in note

### PR DESCRIPTION
**Pull request by sgpinkus**
The term "label" is used extenisively, but (AFAICT) it is never explained what a "label" is. Is an integer offset a label? Do labels apply to rows and columns, multi-indexes? What's the relation between an "index" "axis" and "label" ... etc.

- [ ] closes #xxxx
- [ ] tests added / passed
- [ ] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry

### CODE REVIEW

**Comment by WillAyd** [comment]
I find this note a little confusing. Would just mentioning that a `label` is an object that provides an alternative to positional selection suffice?

**Comment by sam-at-github** [comment]
I agree. It is too verbose, and there is already too many notes and warnings (which I tend to have a habit of not reading anyway :)). I will re-word.

File: doc/source/user_guide/indexing.rst [hunk 1]
```diff
@@ -6,17 +6,18 @@
 Indexing and Selecting Data
 ***************************
 
-The axis labeling information in pandas objects serves many purposes:
+In pandas, data structure axes such as the rows and columns of a DataFrame may have
```
Comment (WillAyd) [inline doc/source/user_guide/indexing.rst]: To be clear, an Index (which is used for selection along an axis) *does* have labels. This is a main thing that separates pandas from libraries like NumPy

**Review by WillAyd — COMMENTED** [review]
Hmm still not sure this is it but cc @datapythonista in case he has any insight to offer

**Comment by datapythonista** [comment]
I agree that we should be more explicit on the terms we used. I think we discussed in the past about having a glossary, which I think it's the right way to address this. I could only find this related issue: #21784

I don't think this PR is helping much. I'll close this, and open an issue for the glossary, feel free to work on that.

**Comment by WillAyd** [comment]
Sounds good. May be loosely intertwined with what we are defining in pandas._typing

@sam-at-github thanks for the PR. To the point above though I think we'd rather make a more comprehensive change than what is offered here. Would certainly value input if you'd like to put thoughts into the other issue

**Comment by datapythonista** [comment]
Sorry, re-reading my last comment, just realized that probably didn't sound really nice... Not my intention.

@sam-at-github thanks for your contribution, and it really helps, since this is very helpful feedback. I opened #27019 since I think a glossary will make it easier to address what you're addressing here. Feel free to continue the work you started here, but I think it makes more sense to close this, and continue in a PR that addresses that issue.

Thanks!

**Comment by sam-at-github** [comment]
Sounds like perfect is the enemy of the good around here. Good luck with glossary.

File: doc/source/user_guide/indexing.rst (hunk 1 above)
Comment (sam-at-github) [reply doc/source/user_guide/indexing.rst]: Yes I know. It's actually quite challenging to unpack axis, index, labels, positional indexation via integers. An index really *is* labels. I found it an improvement over "axis labeling information ... serves many purposes", by being less nebulous and vague.

**Pull request closed without merging**